src/app/core/entity-components/entity-details/entity-details.component.spec.ts

```diff
@@ -151,4 +160,12 @@ describe("EntityDetailsComponent", () => {
     expect(entityMapper.save).toHaveBeenCalledWith(testChild, true);
     expect(router.navigate).toHaveBeenCalled();
   }));
+
+  it("should call router when user is not permitted to create entities", () => {
+    mockEntityPermissionsService.userIsPermitted.and.returnValue(false);
```

Comment:
I don't get the assumption behind this test - if not allowed (to create? or what permission) then you ensure there is navigation to the new form rather than no navigation or navigation to the list/dashboard?

Reply:
As the name of the test says, it is expected that the router is being called if the user opens for example `/child/new` but is not permitted to create new `Child` entities. 

This should prevent users from manually navigating to the `/new` page when the `add` button is disabled on the list view.
For now the test only expects the router to be called, because I did not want to make any further assumptions on which route the router will route to.